specs/CriticalUpdateAvailable.md

```diff
@@ -0,0 +1,106 @@
+Critical Update Available
+===
+
+# Background
+WebView2 developers often have to author ECS configurations to toggle feature flags. However, once 
+these payloads are received, there is no way to restart the WebView2 and apply the payload. The 
+purpose of this API is to detect such critical payloads and inform the developer so they may s
+restart their app or their WebView2 or other appropriate action.
+
+# Examples
+## WinRT and .NET   
+```c#
+void WebView_CoreWebView2InitializationCompleted(object sender, CoreWebView2InitializationCompletedEventArgs e)
+{
+    if (e.IsSuccess)
+    {
+        // ...
+        webView.CoreWebView2.Environment.CriticalRestartRequired += WebView_CriticalRestartRequired;
+    }
+}
+
+void WebView_CriticalRestartRequired(object sender, object e)
+{
+    System.Threading.SynchronizationContext.Current.Post((_) =>
```

Comment:
The C++ and C# code should match. Right now they don't really. The C# uses this post to run this code async where the c++ does not. You don't need to do it in either since RestartIfSelectedByUser could handle this.

The C# code shows an outer method WebView_CoreWebView2InitializationCompleted that registers the event handler and the C++ just has some code that registers the event handler.